packages/replay/src/session/getSession.ts

```diff
@@ -5,59 +5,72 @@ import { createSession } from './createSession';
 import { fetchSession } from './fetchSession';
 import { makeSession } from './Session';
 
-interface GetSessionParams extends SessionOptions {
-  timeouts: Timeouts;
+/**
+ * Check a session, and either return it or a refreshed version of it.
+ * The refreshed version may be unsampled.
+ * You can check if the session has changed by comparing the session IDs.
+ */
+export function maybeRefreshSession(
+  session: Session,
+  {
+    timeouts,
+    traceInternals,
+  }: {
+    timeouts: Timeouts;
+    traceInternals?: boolean;
+  },
+  sessionOptions: SessionOptions,
+): Session {
+  // If not expired, all good, just keep the session
+  if (!isSessionExpired(session, timeouts)) {
+    return session;
+  }
 
-  /**
-   * The current session (e.g. if stickySession is off)
-   */
-  currentSession?: Session;
+  const isBuffering = session.sampled === 'buffer';
 
-  traceInternals?: boolean;
+  // If we are buffering & the session may be refreshed, just return it
+  if (isBuffering && session.shouldRefresh) {
+    return session;
+  }
+
+  // If we are buffering & the session may not be refreshed (=it was converted to session previously already)
+  // We return an unsampled new session
+  if (isBuffering) {
+    logInfoNextTick('[Replay] Session should not be refreshed', traceInternals);
+    return makeSession({ sampled: false });
+  }
+
+  // Else, we are not buffering, and the session is expired, so we need to create a new one
+  logInfoNextTick('[Replay] Session has expired, creating new one...', traceInternals);
+
+  const newSession = createSession(sessionOptions, { previousSessionId: session.id });
+
+  return newSession;
 }
 
 /**
- * Get or create a session
+ * Get or create a session, when initializing the replay.
+ * Returns a session that may be unsampled.
  */
-export function getSession({
-  timeouts,
-  currentSession,
-  stickySession,
-  sessionSampleRate,
-  allowBuffering,
-  traceInternals,
-}: GetSessionParams): { type: 'new' | 'saved'; session: Session } {
+export function loadOrCreateSession(
```

Comment:
l: Should we extract these two exported functions into 2 separate files and get rid of `getSession.ts`?

Reply:
Hmm I also thought about this, not 100% sure. I guess we can, probably cleaner that way!